internal/cli/push.go

```diff
@@ -48,6 +50,23 @@ func init() {
 	})
 }
 
+// instances checks if the user provide instances. If they don't, then we'll
+// pass nil and eihter use the default or whatever is deployed in the cluster.
+func instances(cmd *cobra.Command) (*int32, error) {
+	var i *int32
+	instances, err := cmd.Flags().GetInt32("instances")
+	if err != nil {
+		return i, errors.Wrap(err, "could not read instances parameter")
+	}
+	cmd.Flags().Visit(func(f *pflag.Flag) {
+		if f.Name == "instances" {
+			n := int32(instances)
+			i = &n
```

Comment:
We seem to search here the flag set ?! I do not understand why.
Can we not directly use lines 63-64 without the search ?

Reply:
The user might not have provided the flag.
This PR changes the instances argument to push to a pointer, so we can later decide if we should use the default value, or the value from the cluster.
The user could explicitly push with the default value, e.g. to scale down. So we can't match on the default number:
"1" could be intentional, or just the default from flag.

There are no optional arguments in flag,`Visit()` solves that problem, it visits only those flags that have been set.

Reply:
Ah. Now I understand. Maybe add a small comment about this ?